src/lib/envelope/envelope.ml

```diff
@@ -1,8 +1,37 @@
 open Core
 open Network_peer
+open Module_version
+
+module Sender = struct
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        let version = 1
+
+        type t = Local | Remote of Peer.t [@@deriving sexp, bin_io]
+      end
+
+      include T
+      include Registration.Make_latest_version (T)
+    end
+
+    module Latest = V1
+
+    module Module_decl = struct
+      let name = "envelope_sender"
+
+      type latest = Latest.t
+    end
+
+    module Registrar = Registration.Make (Module_decl)
+    module Registered_V1 = Registrar.Register (V1)
+  end
+
+  include Stable.Latest
```

Comment:
this exposes the `bin_io` code for `Sender.t`, which is probably not what you want

you can write `type t = Stable.Lates.t [@@deriving sexp]` instead

Reply:
I think it doesn't because of the mli -- do you think I should do what you're suggesting instead?

Reply:
oh, I didn't notice the .mli

but still, it would be safer to not have the possibility that someone changes the .mli

I've been doing it the way I suggested, and adding a comment not to add `bin_io`